src/components/Notifications/SystemStatusBanner.tsx

```diff
@@ -0,0 +1,136 @@
+        left: '50%',
+        padding: theme.spacing(2),
+        position: 'fixed',
+        transform: 'translateX(-50%)'
+    },
+    statusContentContainer: {
+        alignItems: 'flex-start',
+        display: 'flex',
+        maxWidth: theme.spacing(131)
+    },
+    statusClose: {
+        alignItems: 'flex-start',
+        display: 'flex',
+        flex: '0 0 auto'
+    },
+    statusIcon: {
+        alignItems: 'center',
+        display: 'flex',
+        flex: '0 0 auto',
+        lineHeight: `${theme.spacing(3)}px`
+    },
+    statusMessage: {
+        flex: '1 1 auto',
+        fontWeight: 'normal',
+        lineHeight: `${theme.spacing(3)}px`,
+        marginLeft: theme.spacing(2),
+        marginRight: theme.spacing(2)
+    }
+}));
+
+interface StatusConstantValues {
+    color: string;
+    IconComponent: typeof SvgIcon;
+}
+
+const statusConstants: Record<StatusString, StatusConstantValues> = {
+    normal: {
+        color: infoIconColor,
+        IconComponent: Info
+    },
+    degraded: {
+        color: warningIconColor,
+        IconComponent: Warning
+    },
+    down: {
+        color: warningIconColor,
+        IconComponent: Warning
+    }
+};
+
+const StatusIcon: React.FC<{ status: StatusString }> = ({ status }) => {
+    const { color, IconComponent } =
+        statusConstants[status] || statusConstants.normal;
+    return <IconComponent htmlColor={color} />;
+};
+
+const RenderSystemStatusBanner: React.FC<{
+    systemStatus: SystemStatus;
+    onClose: () => void;
+}> = ({ systemStatus: { message, status }, onClose }) => {
+    const styles = useStyles();
+    return (
+        <Paper className={styles.statusPaper} elevation={1} square={true}>
+            <div className={styles.statusIcon}>
+                <StatusIcon status={status} />
+            </div>
+            <div className={styles.statusContentContainer}>
+                <Typography variant="h6" className={styles.statusMessage}>
+                    {message}
+                </Typography>
+            </div>
+            <div className={styles.statusClose}>
+                <ButtonBase
+                    aria-label="Close"
+                    className={styles.closeButton}
+                    onClick={onClose}
+                >
+                    <Close fontSize="small" />
+                </ButtonBase>
+            </div>
+        </Paper>
+    );
+};
+
+/** Fetches and renders the system status returned by issuing a GET to
+ * `env.STATUS_URL`. If the status includes a message, a dismissable toast
+ * will be rendered. Otherwise, nothing will be rendered.
+ */
+export const SystemStatusBanner: React.FC<{}> = () => {
+    const systemStatus = useSystemStatus();
+    const [dismissed, setDismissed] = React.useState(false);
+    const onClose = () => setDismissed(true);
+    if (dismissed) {
+        return null;
+    }
+    return (
+        <WaitForData {...systemStatus}>
+            {systemStatus.value.message && (
```

Comment:
Is there a chance that `systemStatus.value.message` will be `undefined`? If so, I'd suggest turning this into a ternary to explicitly render `null` since `undefined` will blow up React at runtime

Reply:
Oh right, I usually do `!!` for this case. I'll update it to a ternary in the following PR. Thanks for catching it.